### DISCUSSION TITLE
Sign got separated with number if the number is a negative value (e.g. -123)

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#106] [General]
Hi Bruce,
I just noticed that for any negative value after parsing, the sign got separated from the number value.

Given the following two test cases:
`SELECT -123.45;`
`SELECT DATEADD(year, -10, '2017/08/25') AS DateAdd;`

The results I got after parsing
Start...
Statement 1:  SELECT - 123.45
        Token type: Keyword     Text: SELECT
        **Token type: Operator    Text: -
        Token type: NumericLiteral      Text: 123.45**

Start...
Statement 1:  SELECT DATEADD ( year , - 10 , '2017/08/25' ) AS DateAdd
        Token type: Keyword     Text: SELECT
        Token type: SystemIdentifier    Text: DATEADD
        Token type: Character   Text: (
        Token type: SystemIdentifier    Text: year
        Token type: Character   Text: ,
        **Token type: Operator    Text: -
        Token type: NumericLiteral      Text: 10**
        Token type: Character   Text: ,
        Token type: StringLiteral       Text: '2017/08/25'
        Token type: Character   Text: )
        Token type: Keyword     Text: AS
        Token type: SystemIdentifier    Text: DateAdd

Noticed that the negative sign "-" was parsed as an operator. In these cases, should the sign be part of the NumericLiteral?
The above results were produced using the latest build.

Regards
Paul

**Comment by bruce-dunwiddie** [comment]
This is currently intentional because it requires additional context to know if it should be a sign as opposed to an operator, when parsing at the token layer.

Consider "SELECT 1 -123.45".

It's easier to handle it as always an operator, with an implicit "0" preceeding.

A statement level parser that is keeping context and has additional parsing rules based on the preceeding context is really the layer that needs to make the translation.

**Reply by cptb** [reply]
Thank you for the prompt reply. 
I totally understand the intention. I just noticed this today and would like to double confirm with you. It's not causing any problem as our internal process does not do any additional processing with the sign and value. They got passed to SQL Server for direct execution.

Thank you again.
